File: src/main/java/kookmin/software/capstone2023/timebank/application/service/auth/model/AuthenticationRequest.java
package kookmin.software.capstone2023.timebank.application.service.auth.model;

import kookmin.software.capstone2023.timebank.domain.model.AccountType;
import kookmin.software.capstone2023.timebank.domain.model.auth.AuthenticationType;
import lombok.Data;

@Data
public abstract class AuthenticationRequest {
    private final AuthenticationType type;
    private final AccountType accountType;

    public AuthenticationRequest(AuthenticationType type, AccountType accountType) {
        this.type = type;
        this.accountType = accountType;
    }
}

File: src/main/java/kookmin/software/capstone2023/timebank/application/service/auth/model/AuthenticationRequest.kt
package kookmin.software.capstone2023.timebank.application.service.auth.model

import kookmin.software.capstone2023.timebank.domain.model.AccountType
import kookmin.software.capstone2023.timebank.domain.model.auth.AuthenticationType
import kookmin.software.capstone2023.timebank.domain.model.auth.SocialPlatformType

sealed class AuthenticationRequest(
    val type: AuthenticationType,
    open val accountType: AccountType? = null,
) {
    data class SocialAuthenticationRequest(
        val socialPlatformType: SocialPlatformType,
        val accessToken: String,
        override val accountType: AccountType?,
    ) : AuthenticationRequest(type = AuthenticationType.SOCIAL, accountType = accountType)

    data class PasswordAuthenticationRequest(
        val username: String,
        val password: String,
        override val accountType: AccountType?,
    ) : AuthenticationRequest(type = AuthenticationType.PASSWORD, accountType = accountType)
}
